src/pytorch_lightning/lite/lite.py

```diff
@@ -65,376 +84,229 @@ class LightningLite(ABC):
+    """
+    if gpus is not None:
+        rank_zero_deprecation(
+            f"Setting `Lite(gpus={gpus!r})` is deprecated in v1.8.0 and will be removed"
+            f" in v1.10.0. Please use `Lite(accelerator='gpu', devices={gpus!r})` instead."
+        )
+    if tpu_cores is not None:
+        rank_zero_deprecation(
+            f"Setting `Lite(tpu_cores={tpu_cores!r})` is deprecated in v1.8.0 and will be removed"
+            f" in v1.10.0. Please use `Lite(accelerator='tpu', devices={tpu_cores!r})` instead."
+        )
+    deprecated_devices_specific_flag = gpus or tpu_cores
+    if deprecated_devices_specific_flag and deprecated_devices_specific_flag not in ([], 0, "0"):
+        if devices:
+            rank_zero_warn(
+                f"The option `devices={devices}` will be ignored and the device specific number"
+                f"{deprecated_devices_specific_flag} will be used instead."
+            )
+
+        if gpus is not None and tpu_cores is not None:
+            rank_zero_warn(
+                f"Both `Lite(gpus={gpus!r}, tpu_cores={tpu_cores!r})` were specified. Please choose only one of"
+                " the two."
+            )
+
+        if accelerator is None:
+            if tpu_cores:
+                accelerator = "tpu"
+            if gpus:
+                accelerator = "cuda"
+
+    return deprecated_devices_specific_flag, accelerator
+
+
+def _to_lite_strategy(strategy: PLStrategy) -> LiteStrategy:
+    """Re-instantiates a PL-Strategy as the corresponding Lite-Strategy."""
+
+    if type(strategy) == PLDDPStrategy:
+        return LiteDDPStrategy(
+            accelerator=strategy.accelerator,
+            parallel_devices=strategy.parallel_devices,
+            cluster_environment=strategy.cluster_environment,
+            checkpoint_io=strategy.checkpoint_io,
+            precision_plugin=_to_lite_precision_plugin(strategy.precision_plugin),
+            process_group_backend=strategy.process_group_backend,
+            timeout=strategy._timeout,
+            **strategy._ddp_kwargs,
+        )
+
+    if type(strategy) == PLDDPSpawnStrategy:
+        return LiteDDPSpawnStrategy(
+            accelerator=strategy.accelerator,
+            parallel_devices=strategy.parallel_devices,
+            cluster_environment=strategy.cluster_environment,
+            checkpoint_io=strategy.checkpoint_io,
+            precision_plugin=_to_lite_precision_plugin(strategy.precision_plugin),
+            process_group_backend=strategy.process_group_backend,
+            timeout=strategy._timeout,
+            start_method=strategy._start_method,
+            **strategy._ddp_kwargs,
         )
 
-    @staticmethod
-    def _get_distributed_sampler(dataloader: DataLoader, **kwargs: Any) -> DistributedSampler:
-        kwargs.setdefault("seed", int(os.getenv("PL_GLOBAL_SEED", 0)))
-        return DistributedSamplerWrapper(dataloader.sampler, **kwargs)
+    if type(strategy) == PLTPUSpawnStrategy:
```

Comment:
classes are singletons and should be checked with is

```suggestion
    if type(strategy) is PLTPUSpawnStrategy:
```

Reply:
done